ipa-core/src/ff/boolean_array.rs

```diff
@@ -556,3 +703,116 @@ impl rand::distributions::Distribution<BA256> for rand::distributions::Standard
         (rng.gen(), rng.gen()).into()
     }
 }
+
+pub struct BooleanArrayBuilder<T>
+where
+    T: ArrayAccess<Output = Boolean>,
+{
+    array: T,
+    index: usize,
+}
+
+impl<T> BooleanArrayBuilder<T>
+where
+    T: ArrayAccess<Output = Boolean> + SharedValue,
+{
+    fn new() -> Self {
+        Self {
+            array: T::ZERO,
+            index: 0,
+        }
+    }
+}
+
+impl<T> ArrayBuilder for BooleanArrayBuilder<T>
+where
+    T: ArrayAccess<Output = Boolean> + Send,
+{
+    type Element = Boolean;
+    type Array = T;
+
+    fn push(&mut self, value: Self::Element) {
+        self.array.set(self.index, value);
+        self.index += 1;
+    }
+
+    fn build(self) -> Self::Array {
+        self.array
+    }
+}
+
+#[cfg(all(test, unit_test))]
+mod tests {
+    use rand::{thread_rng, Rng};
+
+    use super::*;
+
+    // It does not seem worth running these tests for every BA type, although
+    // it would be worth writing a version for the largest BA type (which
+    // requires replacing truncate_from).
```

Comment:
We could add the test to the macro if we want to test it for any BA type. I would also test it for BA types that are not powers of 2, i.e. BA3. Otherwise I am fine with not testing it for any BA type.

Reply:
I will move it into the macro, it should be fast and that seems easier than introducing a different macro to generate the tests for particular configs.